interface/repository/pkmn_repository.go

```diff
@@ -15,36 +18,68 @@ type pokemonRepository struct {
 	FilePath string
 }
 
+var wg sync.WaitGroup
```

Comment:
You would like to keep these within the context of a request, not globally. If you get multiple requests at the same time to this endpoint, then all of those requests will share this variables and things will get messed up, you would have to pass these on functions.

Maybe the lock can be here since there is only one file for all the requests, but the wait group would definitely have to be one per request